Sources/PackageCollections/Storage/FilePackageCollectionsSourcesStorage.swift

```diff
@@ -48,63 +34,51 @@ struct FilePackageCollectionsSourcesStorage: PackageCollectionsSourcesStorage {
     }
 
     func list() async throws -> [PackageCollectionsModel.CollectionSource] {
```

Comment:
There are no `async` calls in this and other methods, maybe `async` effect and locking from it should be removed, while the type itself made an `actor`?

Reply:
I have two concerns with this. 
1- It looks like motivating reason for the callbacks was parallel decoding of large collections. Maybe that is nontrivial necessary and was premature optimization. 
2- This could be made an actor, but it does not solve synchronization between processes. If more than one process is accessing the same storage on disk then the synchronization needs to be done with file locks not swift concurrency isolation

Reply:
Ah, I missed those were file locks, thanks for the clarification